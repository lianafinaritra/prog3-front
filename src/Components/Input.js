
export default function Input( props ) {
    const { refText, handleLoad } = props;

    return(
        <div className="input-form" data-testid="input-container">
            <div className="relative z-0 w-full mb-6 group">
                <input type="text" name="floating_first_name" id="floating_first_name"
                    className="block py-2.5 px-0 w-full text-sm text-gray-900 bg-transparent border-0 border-b-2 border-gray-300 appearance-none dark:text-white dark:border-gray-600 dark:focus:border-blue-500 focus:outline-none focus:ring-0 focus:border-blue-600 peer"
                    placeholder=" " required ref={refText} data-testid="input" />
                <label htmlFor="floating_first_name"
                    className="block peer-focus:font-medium absolute text-sm text-gray-200 dark:text-gray-900 duration-300 transform -translate-y-6 scale-75 top-3 -z-10 origin-[0] peer-focus:left-0 peer-focus:text-blue-600 peer-focus:dark:text-blue-500 peer-placeholder-shown:scale-100 peer-placeholder-shown:translate-y-0 peer-focus:scale-75 peer-focus:-translate-y-6">Description
                </label>
            </div>
            <button type="submit" id="close" onClick={handleLoad} data-testid="submit"
                className="text-gray-500 bg-gray-200 border border-gray-300 focus:outline-none hover:bg-gray-100 focus:ring-4 focus:ring-gray-200 font-medium rounded-lg text-sm px-5 py-1 mr-1 mb-1 dark:bg-gray-800 dark:text-white dark:border-gray-600 dark:hover:bg-gray-700 dark:hover:border-gray-600 dark:focus:ring-gray-700 ml-6">Ajouter
            </button>
        </div>
    )
}