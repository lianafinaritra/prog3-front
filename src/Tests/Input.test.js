import React from "react";
import {render} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Input from "../Components/Input";
import renderer from "react-test-renderer";

describe('Input test', () => {
    const handleLoad = jest.fn();

    it('renders inputs element', () => {
        const { getByTestId } = render(<Input value={""} handleLoad={handleLoad} />);
        expect(getByTestId('input-container')).toBeInTheDocument();
        expect(getByTestId('input')).toBeInTheDocument();
        expect(getByTestId('submit')).toBeInTheDocument();
    });

    it('write new todo', () => {
        const { getByTestId } = render(<Input value={""} handleLoad={handleLoad} />)

        const newTask = 'new task';
        const input = getByTestId("input");
        expect(input.value).toBe("");
        userEvent.type(input, newTask);
        expect(input.value).toBe(newTask);
    });

    it('call handleLoad when click on the button', () => {
        const { getByTestId } = render(<Input value={""} handleLoad={handleLoad} />);

        const submit = getByTestId('submit');
        userEvent.click(submit);
        expect(handleLoad).toHaveBeenCalled();
    });

    it("should renders correctly", ()=>{
        const inputSnap = renderer.create(
            <Input value={""} handleLoad={(e) => e.preventDefault()} />
        )

        const input = inputSnap.toJSON();
        expect(input).toMatchSnapshot();
    })
});
