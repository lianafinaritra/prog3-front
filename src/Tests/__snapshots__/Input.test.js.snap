// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test should renders correctly 1`] = `
<div
  className="input-form"
  data-testid="input-container"
>
  <div
    className="relative z-0 w-full mb-6 group"
  >
    <input
      className="block py-2.5 px-0 w-full text-sm text-gray-900 bg-transparent border-0 border-b-2 border-gray-300 appearance-none dark:text-white dark:border-gray-600 dark:focus:border-blue-500 focus:outline-none focus:ring-0 focus:border-blue-600 peer"
      data-testid="input"
      id="floating_first_name"
      name="floating_first_name"
      placeholder=" "
      required={true}
      type="text"
    />
    <label
      className="block peer-focus:font-medium absolute text-sm text-gray-200 dark:text-gray-900 duration-300 transform -translate-y-6 scale-75 top-3 -z-10 origin-[0] peer-focus:left-0 peer-focus:text-blue-600 peer-focus:dark:text-blue-500 peer-placeholder-shown:scale-100 peer-placeholder-shown:translate-y-0 peer-focus:scale-75 peer-focus:-translate-y-6"
      htmlFor="floating_first_name"
    >
      Description
    </label>
  </div>
  <button
    className="text-gray-500 bg-gray-200 border border-gray-300 focus:outline-none hover:bg-gray-100 focus:ring-4 focus:ring-gray-200 font-medium rounded-lg text-sm px-5 py-1 mr-1 mb-1 dark:bg-gray-800 dark:text-white dark:border-gray-600 dark:hover:bg-gray-700 dark:hover:border-gray-600 dark:focus:ring-gray-700 ml-6"
    data-testid="submit"
    id="close"
    onClick={[Function]}
    type="submit"
  >
    Ajouter
  </button>
</div>
`;
